test/extended/util/framework.go

```diff
@@ -223,6 +226,38 @@ var CheckDeploymentFailedFunc = func(d *kapi.ReplicationController) bool {
 	return d.Annotations[deployapi.DeploymentStatusAnnotation] == string(deployapi.DeploymentStatusFailed)
 }
 
+// GetRunningPodNames looks up pods that are in a Running state and returns their names.
+func GetRunningPodNames(c kclient.PodInterface, label labels.Selector) (podNames []string, err error) {
```

Comment:
Make this generic function that accept a ConditionFunc as argument. Then use the ConditionFunc to determine what pods to return....

GetPodNamesByFilter(PodInterface, labels, conditionFunc).